lib/Importer/Caffe2ModelLoader.cpp

```diff
@@ -849,6 +849,98 @@ llvm::Error Caffe2ModelLoader::loadOperator(const caffe2::OperatorDef &op) {
     return llvm::Error::success();
   }
 
+  if (typeName == "SparseLengthsSumFused8BitRowwise" ||
+      typeName == "SparseLengthsWeightedSumFused8BitRowwise") {
+    // If SparseLengthsWeightedSumFused8BitRowwise, then the weights are the
+    // second input and so we need to shift indices/lengths.
+    size_t indicesIdx = 1;
+    size_t lengthsIdx = 2;
+    if (typeName == "SparseLengthsWeightedSumFused8BitRowwise") {
+      indicesIdx++;
+      lengthsIdx++;
+    }
+
+    NodeValue data;
+    ASSIGN_VALUE_OR_RETURN_ERR(data,
+                               getNodeValueOrCreateConstantByName(op.input(0)));
+    NodeValue indices;
+    ASSIGN_VALUE_OR_RETURN_ERR(
+        indices, getNodeValueOrCreateConstantByName(op.input(indicesIdx)));
+    NodeValue lengths;
+    ASSIGN_VALUE_OR_RETURN_ERR(
+        lengths, getNodeValueOrCreateConstantByName(op.input(lengthsIdx)));
+
+    // Each row of data stores quantized values, and then 4-byte scale and
+    // 4-byte bias. Thus data must be a Constant so we can retrieve the
+    // Scales/Offsets inputs for RowwiseQuantizedSparseLengthsWeightedSumNode.
+    Constant *fusedDataC = llvm::dyn_cast<Constant>(data);
+    RETURN_ERR_IF_NOT(
+        fusedDataC,
+        "Input data must be Constant in order to retrieve Scales/Offsets.");
+    RETURN_ERR_IF_NOT(fusedDataC->getElementType() == ElemKind::Int8QTy,
+                      "Data must be Int8QTy.");
+    auto dataDims = fusedDataC->dims();
+    static_assert(sizeof(float) == 4, "float must be 4 bytes.");
+    RETURN_ERR_IF_NOT(dataDims.size() == 2 && dataDims[1] > 8,
+                      "Fused data must be 2 dimemsional, with 8 extra columns "
+                      "for holding the per row float scale and offset");
```

Comment:
nit: float scale and int32 offset.

Reply:
They're actually both float in Caffe2 -- [e.g. see here](https://caffe2.ai/docs/operators-catalogue.html#floattofused8bitrowwisequantized):

>bias as a 32-bit float

Reply:
Got it, interesting. other C2 quantized ops assumes different.